docs/rfc/rfc-027-p2p-message-bandwidth-report.md

```diff
@@ -0,0 +1,137 @@
+
+#### Compress Transmitted Data
+
+Tendermint's data is sent uncompressed on the wire. The messages are not compressed before sending and the transport performs no compression either. Some of the information communicated by Tendermint is a poor candidate for compression: Data such as digital signatures and hashes have high entropy and therefore do not compress well. However, transactions may contain lots of information that has less entropy. Compression within Tendermint may be added at several levels. Compression may be performed at the [Tendermint 'packet' level][must-wrap-packet] or at the [Tendermint message send level][message-send].
+
+#### Transmit Less Data During Block Gossip
+
+Block, vote, and mempool gossiping transmit much of same data. The mempool reactor gossips candidate transactions to each peer. The consensus reactor, when gossiping the votes, sends vote metadata and the digital signature of that signs over that metadata. Finally, when a block is proposed, the proposing node amalgamates the received votes, a set of transaction, and adds a header to produce the block. This block is then serialized and gossiped as a list of bytes. However, the data that the block contains, namely the votes and the transactions were most likely _already transmitted to the nodes on the network_ via mempool transaction gossip and consensus vote gossip. 
+
+Therefore, block gossip can be updated to transmit a representation of the data contained in the block that assumes the peers will already have most of this data. Namely, the block gossip can be updated to only send 1) a list of transaction hashes and 2) a bit array of votes selected for the block along with the header and other required block metadata.
+
+This new proposed method for gossiping block data would require a slight update to the mempool transaction gossip and consensus vote gossip. Since all of the contents of each block will not be gossiped together, it's possible that some nodes are missing a proposed transaction or the vote of a validator indicated in the new block gossip format. The mempool and consensus reactors would need to be updated to provide a `NeedTxs` and `NeedVotes` message. Each of these messages would allow a node to request a set of data from their peers. When a node receives one of these, it will then transmit the Tx/Votes indicate in the associated message regardless of whether it believes it has transmitted them to the peer before.
```

Comment:
Just to point it out: the Narwhal mempool operation is similar to the one proposed here. Meaning, we don't broadcast txs in a block, but only tx ids. If you search for Narwhal on the shared-tendermint slack channel, you will find discussions about that.

Reply:
>  If the gossip network is working properly, you will eventually receive all the data you need.

You're completely correct. My worry, which turns out to be false, was if a peer crashes than we may not broadcast them the data that they require since we would already have marked them as having received it. However, we reinitialize the peer state each time it connects, and therefore create a new view of the peer's state, rendering this not a problem.

Reply:
Added discussion of this here @sergio-mena

Reply:
🙏